### DISCUSSION TITLE
How set focus

### DISCUSSION THREAD

**Discussion by jgracia** [sa-si-dev/virtual-select#79] [Q&A] [answered]
How to set focus programmatically.

**Comment by sa-si-dev** [comment]
@jgracia Please use the below function to focus programmatically. I will add a dedicated method in the plugin itself in the future release.

```js
focusVirtualSelect('sample-select');

function focusVirtualSelect(id) {
  var $ele = document.querySelector('#' + id);

  if ($ele) {
    var $wrapper = $ele.querySelector('.vscomp-ele-wrapper');

    if ($wrapper) {
      $wrapper.focus();
    }
  }
}
```

**Reply by jgracia** [reply]
Thank you very kind. I will prove it.

**Comment by sa-si-dev** [comment, marked as answer]
@jgracia focus() method has been added in the [release v1.0.19](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.19)

It could be used as below.

```js
document.querySelector('#sample-select').focus();
```
